### PR TITLE
[CARBONDATA-1477] Corrected the Bug For Wrong Values Of Date data type in hive

### PR DESCRIPTION
**Pull request by anubhav100**
https://issues.apache.org/jira/browse/CARBONDATA-1477

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/106/

**Comment by anubhav100** [comment]
@chenliang613 @cenyuhai

**Comment by ravipesala** [comment]
Build Failed with Spark 1.6, Please check CI http://144.76.159.231:8080/job/ApacheCarbonPRBuilder1/9/

File: integration/hive/src/main/java/org/apache/carbondata/hive/CarbonDictionaryDecodeReadSupport.java [hunk 1]
```diff
@@ -228,9 +229,13 @@ private Writable createWritablePrimitive(Object obj, CarbonColumn carbonColumn)
       case LONG:
         return new LongWritable((long) obj);
       case SHORT:
-        return new ShortWritable((Short) obj);
+        return new ShortWritable((short) obj);
       case DATE:
-        return new DateWritable(new Date(((Integer) obj).longValue()));
+        Calendar c = Calendar.getInstance();
+        c.setTime(new Date(0));
```
Comment (cenyuhai) [inline integration/hive/src/main/java/org/apache/carbondata/hive/CarbonDictionaryDecodeReadSupport.java]: Does it mean current time? and then reset the year to current year?

Comment (anubhav100) [reply integration/hive/src/main/java/org/apache/carbondata/hive/CarbonDictionaryDecodeReadSupport.java]: Yup

Comment (anubhav100) [reply integration/hive/src/main/java/org/apache/carbondata/hive/CarbonDictionaryDecodeReadSupport.java]: There are changes in master code for date so these changes are required for getting the right values at hive side

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/735/

**Comment by anubhav100** [comment]
@chenliang613 can you review this pr?

**Comment by anubhav100** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/159/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/788/

**Comment by anubhav100** [comment]
Retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 1.6, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/53/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/179/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/807/

**Comment by chenliang613** [comment]
@anubhav100 can you provide data.csv file which you tested, let me verify it.

**Comment by anubhav100** [comment]
@chenliang613 I attached the csv with same JIRA issue https://issues.apache.org/jira/browse/CARBONDATA-1477

**Comment by chenliang613** [comment]
LGTM

**Pull request closed without merging**